### DISCUSSION TITLE
Can/should actions/cache be used for publishing?

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#4317] [Q&A]
Using a GitHub Action to publish changes to a gh-pages branch and was now curious if it would be worth it to implement and use `actions/cache` for possible faster runs.

If it would be better, what is the recommended setup to achieve optimal caching while still using the latest versions (I use Dependabot to keep MkDocs and Material for MkDocs updated)?

**Comment by squidfunk** [comment]
MkDocs doesn't cache its output. The site directory is cleaned when a build is started. The only caches that make sense, and which we also advertise on the documentation are related to the [privacy](https://squidfunk.github.io/mkdocs-material/setup/ensuring-data-privacy/#caching) (to safe time going onto the network) and [social](https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#caching) (to save time generating images) plugins, which write their output to `.cache`.

**Reply by Andre601** [reply]
I'm asking in a more general sense, as in caching pip and dependencies as a whole when used in an action.

**Reply by squidfunk** [reply]
Other than that, if you like to cache Python dependencies, you can compute a hash from a lockfile, i.e. the output of `pip freeze`. This will also give you reproducible builds, which makes a lot of sense. However, we did this some time ago and ended up with stale caches. Thus, it should be done with caution.

**Reply by Andre601** [reply]
How should I understand a stale cache in a python-sense?
From what I've read does the `setup-python` action have a `cache` action to cache dependencies using the hashed `requirements.txt` file as its key.
